lib/reporters/base.js

```diff
@@ -211,6 +211,56 @@ var generateDiff = (exports.generateDiff = function (actual, expected) {
   }
 });
 
+/**
+ * Traverses err.cause and returns all stack traces
+ *
+ * @private
+ * @param {Error} err
+ * @param {Set<Error>} [seen]
+ * @return {{ message: string, msg: string, stack: string }}
+ */
+var getFullErrorStack = function (err, seen) {
+  if (seen && seen.has(err)) {
```

Comment:
[Style] Nit: since Mocha v10 supports Node >=14:

```suggestion
  if (seen?.has(err)) {
```